### ISSUE TITLE
Unable to access jarfile IPMonitor.jar

### ISSUE THREAD

**Issue by AppleTorrent** [pupi1985/IPMonitor#13]
IPMonitor-master/IPMonitor/launchers/macos-admin-gui-launcher.command ; exit;
0:71: execution error: Error: Unable to access jarfile IPMonitor.jar (1)
logout
Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

Seems like the file is not in the directory.

**Comment by pupi1985** [comment]
I see it. It is right there. Please, download it from the releases section or provide much more information about the issue.
![image](https://user-images.githubusercontent.com/1449790/36825675-7f79e5c4-1ce7-11e8-9a2c-478154c9d31f.png)

**Comment by AppleTorrent** [comment]
Works fine now. Thank you. By the way, any chance i can force the app to decrease the interval to 1 minute?

**Comment by pupi1985** [comment]
Check this: #4

**Comment by AppleTorrent** [comment]
Awesome, will check on that later. Thank you so much for your help.

**Comment by AppleTorrent** [comment]
Hey, I found this website https://ipify.org they doesn't have any limitation on their IP request. 
"You can use it without limit (even if you're doing millions of requests per minute)."
I have changed "checking URL" setting on mine. Also tried changing the value of "minCheckingInterval" to 1, but 10 minute limitation still applied. Seems like I didn't find the correct one. Any chance you can guide me to forcefully change the minimum interval to 1 sec?

**Comment by pupi1985** [comment]
1 sec is extremely slow. A request won't finish and you will firing another one. I'd use 60 seconds, at minimum.

You set it here: https://github.com/pupi1985/IPMonitor/blob/6b766d3ce778a0e969e48d4db1e929cd13fcc986/IPMonitor/src/model/configuration/ConfigurationManager.java#L43

And then it is used here: https://github.com/pupi1985/IPMonitor/blob/6b766d3ce778a0e969e48d4db1e929cd13fcc986/IPMonitor/src/model/ipmonitor/IPMonitor.java#L94-L96

You could also comment the exception throw on line 95.

Obvious question: have you recompiled the project? :)

**Comment by AppleTorrent** [comment]
Yeah I know, It'll only change the minimum time interval right? Since probably i can just set it to 60 secs on the settings later.
By the way, why your extension is .java, while mine is .class I'm also facing some trouble on editing the .class file a simple normal editor won't read it, it came out as a hex code i believe or decimal? 
I did recompiled the project ^_^

**Comment by pupi1985** [comment]
Hmm... I'm getting the impression you're not a developer :) That will considerably complicate the process of compiling the project. Explaining how to recompile the project is not trivial and will take a lot of time for me to write and provide support about it.

But the steps would be:

1. Download/checkout the code repository (not the files in the release section but rather from the main page: [or from here](https://github.com/pupi1985/IPMonitor/archive/master.zip))
1. Import the project using an IDE (the project is in Eclipse IDE so better to use that tool and import the project from there)
1. Update the .java files as I explained above
1. Build the project
1. Replace the `IPMonitor.jar` file with the newly created one

Google and read a lot on how to perform steps 2 and 4.

**Comment by AppleTorrent** [comment]
unfortunately yes, I don't have any coding experience. Now I'm on step 3. You said something have to be changed on IPMonitor/IPMonitor/src/model/ipmonitor/IPMonitor.java ? is that a must? I don't know what should i change there.

**Comment by pupi1985** [comment]
Nothing. Just change`600` to `60` and that should do the trick.

**Comment by AppleTorrent** [comment]
Awesome all working fine now.
https://drive.google.com/file/d/1lNOCt32FxO8MrRKRWHKdpi0gsOqm6U8y/view?usp=sharing
I tried to make it as an app, so they doesn't have to run it from "macos-admin-gui-launcher.command" anymore. You can use if if you want. 
Anyway thank you so much for your help. ^_^

**Comment by pupi1985** [comment]
Well, you solve it quite fast! I use Linux Debian so I don't have a Mac so I can't try that. Also it contains the IP Monitor itself inside of it which won't allow it to run as a daemon (I'm guessing that, actually) and it will force me to make a separate release for Mac, which, as I don't really have one, I can't easily support it.

Anyway, I'm happy that you've managed to adapt it to your needs :+1:

**Comment by AppleTorrent** [comment]
Took an hour to find the "Build Project" on the menu. I kept exporting it from "File" drop down menu, that's why it wasn't working.
Hmm I'm not sure about the daemon actually. But basically it's still a same app just in a different container. On my side, so far so good~
Reason I make it into an app is just because it's a little bit annoying, that every time I ran the ".command" file, I have to type my password in. When I turn it into an app, it doesn't asking password anymore. And it looks nicer! hahahaha. since i can store it at the application folder and then run it from launchpad (place to browse all application in application folder.) like any other app.
